test/api_integration/apis/data_views/swap_references/main.ts

```diff
@@ -0,0 +1,155 @@
+ * Side Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+import {
+  DATA_VIEW_SWAP_REFERENCES_PATH,
+  SPECIFIC_DATA_VIEW_PATH,
+  DATA_VIEW_PATH,
+} from '@kbn/data-views-plugin/server';
+import { ELASTIC_HTTP_VERSION_HEADER } from '@kbn/core-http-common';
+import { INITIAL_REST_VERSION } from '@kbn/data-views-plugin/server/constants';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const title = 'logs-*';
+  const prevDataViewId = '91200a00-9efd-11e7-acb3-3dab96693fab';
+  let dataViewId = '';
+
+  describe('main', () => {
+    const kibanaServer = getService('kibanaServer');
+    before(async () => {
+      const result = await supertest
+        .post(DATA_VIEW_PATH)
+        .send({ data_view: { title } })
+        .set(ELASTIC_HTTP_VERSION_HEADER, INITIAL_REST_VERSION);
+      dataViewId = result.body.data_view.id;
+    });
+    after(async () => {
+      await supertest
+        .delete(SPECIFIC_DATA_VIEW_PATH.replace('{id}', dataViewId))
+        .set(ELASTIC_HTTP_VERSION_HEADER, INITIAL_REST_VERSION);
+    });
+    beforeEach(async () => {
+      await kibanaServer.importExport.load(
+        'test/api_integration/fixtures/kbn_archiver/saved_objects/basic.json'
+      );
+    });
+    afterEach(async () => {
+      await kibanaServer.importExport.unload(
+        'test/api_integration/fixtures/kbn_archiver/saved_objects/basic.json'
+      );
+    });
+
+    it('can preview', async () => {
+      const res = await supertest
+        .post(DATA_VIEW_SWAP_REFERENCES_PATH)
+        .set(ELASTIC_HTTP_VERSION_HEADER, INITIAL_REST_VERSION)
+        .send({
+          from_id: prevDataViewId,
+          to: dataViewId,
+        });
+      expect(res).to.have.property('status', 200);
+    });
+
+    it('can preview specifying type', async () => {
+      const res = await supertest
+        .post(DATA_VIEW_SWAP_REFERENCES_PATH)
+        .set(ELASTIC_HTTP_VERSION_HEADER, INITIAL_REST_VERSION)
+        .send({
+          from_id: prevDataViewId,
+          from_type: 'index-pattern',
+          to: dataViewId,
+        });
+      expect(res).to.have.property('status', 200);
+    });
+
+    it('can save changes', async () => {
+      const res = await supertest
+        .post(DATA_VIEW_SWAP_REFERENCES_PATH)
+        .set(ELASTIC_HTTP_VERSION_HEADER, INITIAL_REST_VERSION)
+        .send({
+          from_id: prevDataViewId,
+          to: dataViewId,
+          preview: false,
+        });
+      expect(res).to.have.property('status', 200);
+
+      const res2 = await supertest
+        .post(DATA_VIEW_SWAP_REFERENCES_PATH)
+        .set(ELASTIC_HTTP_VERSION_HEADER, INITIAL_REST_VERSION)
+        .send({
+          from_id: dataViewId,
+          to: dataViewId,
+        });
+
+      expect(res2.body.result.length).to.equal(1);
```

Comment:
This part does not save changes, right? Does it fit to the test description?

Reply:
I removed this.